### DISCUSSION TITLE
Tips for speeding up qodana on Gitlab CI

### DISCUSSION THREAD

**Discussion by josephtaylor** [JetBrains/Qodana#248] [Q&A]
Does anyone have any tips for speeding up Qodana on Gitlab CI? We're specifically running it on java / spring-boot services. The "project configuration" stage often takes up to 5 minutes, while the analysis generally doesn't take that long (30s to a minute).

What is the configuration stage doing? Is it running a maven build? Is it executing tests in said build?

**Comment by brichbash** [comment]
Hello,

The configuration stage in the JVM linter mostly involves downloading Gradle or Gradle/Maven dependencies. In some cases, it also includes downloading JDK if your project requires JDK different from what is bundled in the Docker image. The stage also includes indexation, which is required for the linters to be able to resolve different symbols across the source code. And no, it doesn't execute tests, it's performing a project sync which is loading your Gradle/Maven settings and their dependencies and building your project model based on the used build system.

However, all of those activities are when using caching, given that there were no significant changes in the project structure and dependencies. At the moment, almost all CI/CD platforms described [in our documentation](https://www.jetbrains.com/help/qodana/ci.html) allow caching, and you can find an example configuration on the corresponding CI/CD platform page.

While I get that probably it's not relevant to the topic starter now, as a year has passed (sorry for missing your question), hopefully, it explains the mechanism.
